dropshot/src/server.rs

```diff
@@ -467,6 +712,38 @@ impl<C: ServerContext> Service<Request<Body>> for ServerRequestHandler<C> {
     }
 }
 
+fn error(err: String) -> std::io::Error {
+    std::io::Error::new(std::io::ErrorKind::Other, err)
+}
+
+// Load public certificate from file.
+fn load_certs(filename: &str) -> std::io::Result<Vec<rustls::Certificate>> {
```

Comment:
Nitpick: I recommend using a `&Path` object for the filenames - there's an example here: https://github.com/tokio-rs/tls/blob/master/tokio-rustls/examples/server/src/main.rs#L33-L41

(Parsing it in the config as a `PathBuf`, as they did, also seems reasonable).

Reply:
And not `P: AsRef<Path>` (possibly just forwarding to an internal &Path version)? Just curious where y'all have fallen on that choice.

Reply:
Oh, that works great too! That basically just adds flexibility, but yeah, making it explicit that this is a path instead of an arbitrary string is a good sanitization choice.

(technically the "as ref" version is the most flexible, but it matters less for a non-pub function)